### PR TITLE
JNG-5852 transform fallback dashboard

### PR DESCRIPTION
**Pull request by noherczeg**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-5852" title="JNG-5852" target="_blank">Sub-taskJNG-5852</a>  Transform fallback dashboard
  </td></table>
  <br />
 

- https://blackbelt.atlassian.net/browse/JNG-5852
- https://blackbelt.atlassian.net/browse/JNG-5853
- https://blackbelt.atlassian.net/browse/JNG-5855

### CODE REVIEW

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/structure/transferDeclaration.etl [hunk 1]
```diff
@@ -43,3 +43,41 @@ rule ClassType
...
+        var label = s.getLabelModifier();
+        if (label.isDefined()) {
+            t.label = label.value.value;
+        }
+        t.titleFrom = UI!ui::TitleFrom#LABEL;
+        t.type = UI!ui::PageContainerType#VIEW;
+        // t.onInit = s.equivalent("TransferObjectTableAddSelectorTableRangeActionDefinition");
+        actorDeclaration.equivalent("Application").pageContainers.add(t);
+        t.children.add(s.equivalent("TransferDeclarationVisualElement"));
+        // t.actionButtonGroup = s.equivalent("TransferObjectTableAddSelectorButtonGroup");
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/structure/transferDeclaration.etl:64]: are these intentionally commented?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewFieldDeclaration.etl [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+rule BooleanTypeCheckbox
+    transform s: JSL!ViewFieldDeclaration
+    to t: UI!ui::Checkbox
+    extends AbstractViewFieldDeclaration {
+        guard: s.referenceType.`primitive`.isDefined()
+            and s.referenceType.`primitive` == "boolean"
+            // and (s.booleanWidget.isUndefined() or s.booleanWidget <> ESM!esm::ui::BooleanWidget#COMBO)
+
+        s.eContainer.uiContainer().children.add(t);
+        t.setId(actorDeclaration.name + "/(jsl/" + s.getId() + ")/BooleanTypeTrinaryLogicCombo");
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewFieldDeclaration.etl]: wrong id?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewFieldDeclaration.etl [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+        var id = actorType.name + "/(esm/" + s.getId() + ")/EnumerationTypeRadio";
+        t.setId(actorType.name + "/(esm/" + s.getId() + ")/EnumerationTypeRadio");
+        for (m in s.dataFeature.getMember().getDataType().members) {
+            t.options.add(m.equivalentDiscriminated("EnumerationMemberOption", id));
+        }
+
+        s.eContainer.uiContainer().children.add(t);
+        log.debug("EnumerationTypeRadio: " + t.name);
+}
+*/
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewFieldDeclaration.etl:131]: Is this intentionally commented?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewFieldDeclaration.etl [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    to t: UI!ui::NumericInput
+    extends AbstractViewFieldDeclaration {
+        guard: s.referenceType.`primitive`.isDefined()
+            and s.referenceType.`primitive` == "numeric"
+
+        s.eContainer.uiContainer().children.add(t);
+        t.setId(actorDeclaration.name + "/(jsl/" + s.getId() + ")/NumericTypeVisualInput");
+        // if (s.formatValue.isDefined()) {
+        //     t.formatValue = s.formatValue;
+        // }
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewFieldDeclaration.etl:161]: comments?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewFieldDeclaration.etl [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+    extends AbstractViewFieldDeclaration {
+        guard: s.referenceType.`primitive`.isDefined()
+            and s.referenceType.`primitive` == "string"
+            and (not s.getLines().isDefined() or s.getLines().value == 1)
+
+        s.eContainer.uiContainer().children.add(t);
+        t.setId(actorDeclaration.name + "/(jsl/" + s.getId() + ")/NumericTypeVisualInput");
+        // if (s.isTypeAheadField.isDefined() and s.isTypeAheadField == true) {
+        //     t.isTypeAheadField = true;
+        // }
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewFieldDeclaration.etl:178]: comments?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewFieldDeclaration.etl [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+rule StringTypeTextInput
+    transform s: JSL!ViewFieldDeclaration
+    to t: UI!ui::TextInput
+    extends AbstractViewFieldDeclaration {
+        guard: s.referenceType.`primitive`.isDefined()
+            and s.referenceType.`primitive` == "string"
+            and (not s.getLines().isDefined() or s.getLines().value == 1)
+
+        s.eContainer.uiContainer().children.add(t);
+        t.setId(actorDeclaration.name + "/(jsl/" + s.getId() + ")/NumericTypeVisualInput");
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewFieldDeclaration.etl]: wrong id

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewFieldDeclaration.etl [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+                and not s.dataFeature.getMember().isQuery
+                and s.dataFeature.getMember().getDataType().isTypeOf(ESM!esm::type::StringType)
+                and s.textWidget == ESM!esm::ui::TextWidget#TEXT
+            s.eContainer.uiContainer().children.add(t);
+            t.format = "%s";
+            t.isMultiLine = s.textMultiLine;
+            t.setId(actorType.name + "/(esm/" + s.getId() + ")/StringTypeFormatted");
+            log.debug("StringTypeFormatted: " + t.name);
+}
+*/
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewFieldDeclaration.etl:213]: comment?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/structure/transferDeclaration.etl (hunk 1 above)
Comment (noherczeg) [reply judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/structure/transferDeclaration.etl:64]: Yes, all of them.

**Review by bencegelei — APPROVED** [review]

**Pull request merged**